### ISSUE TITLE
project Ban Hammer

### ISSUE THREAD

**Issue by BerkeleyTrue** [FreeCodeCamp/camperbot#5]
# Project Ban Hammer

In order to facilitate troll extermination we need a system to promote campers to admins in Gitter without giving them write access on Free Code Camp org. We can do this. We have the power.

This will come in three parts.

* Promote CamperBot's github user to an owner on FCC org.
* Create a common where camperbot will enter the text `/ban @<someUser>` in all our main chat rooms. 
* Checks if the user giving the command is a FCC admin (This will be a list on the server that will not be tracked here on the Repo)

**Comment by SaintPeter** [comment]
I propose `!ban` rather than `/ban`, so it doesn't conflict with the actual `/ban` command.

**Comment by BerkeleyTrue** [comment]
The command can be what ever we want. We can even do 'fluffy foo @<username>`

**Comment by SaintPeter** [comment]
> We can even do `fluffy foo @<username>`

I fully endorse and insist on this command name.

**Comment by joepurdy** [comment]
I humbly submit `bye felicia @<username>` as a candidate for the ban command

**Comment by SaintPeter** [comment]
**Project Ban Hammer, Test 0001**
Naive implementation, have the bot "say" `/ban @username` appears to have failed.  The bot literally says it, no effect on test subject.  

Looking through the docs it's not clear the a bot even *can* issue commands as distinct from messages.

**Comment by BerkeleyTrue** [comment]
The bot cannot ban people until it has write access.

**Comment by SaintPeter** [comment]
I have my personal bot, "SaintBot", set up with a room created by that account.  I was able to manually ban a user with that account, but the bot was not able to with the same account.

**Comment by BerkeleyTrue** [comment]
Dang. Does the text show up in the general chat?

**Comment by SaintPeter** [comment]
Yes: :point_up: [November 22, 2015 6:20 PM](https://gitter.im/SaintBot/testing?at=565277ff54ba2ca242611ca7)

Note that although the activity shows a ban and unban in that room, that was me confirming that I as "SaintBot" could manually ban/unban them.

@abhisekp had a conversation with @madlittlemods and he indicated he didn't know if it would work or not and asked us to file an issue:  gitterHQ/gitter#989

So, at least for the moment, this appears to be a dead letter.

**Comment by BerkeleyTrue** [comment]
Bummer.

**Comment by ltegman** [comment]
According to a recent response on that issue this functionality should be possible through a previously undocumented API endpoint.

**Comment by abhisekp** [comment]
@ltegman yes. It's possible using the following API command endpoints
### Ban
```text
POST /rooms/:roomId/bans
{"username":"trolluser"}
```
### Unban
```text
DELETE /rooms/:roomId/bans/:username
```

**Comment by sludge256** [comment]
Since we have a [repo for Gitter mods](https://github.com/FreeCodeCamp/gitter-mods) and the recent ability to [add mods directly through Gitter](https://gitter.im/gitterHQ/topics/topic/5809efc7b17b667961e0b359/moderation-tools) (even though we've decided to stick with using the repo) would we consider this issue to be resolved and no longer necessary?
Having camperbot issue the ban also raises some issues with logging and resolving ban cases.

**Comment by BerkeleyTrue** [comment]
The only thing we can't do now that we could program camperbot to do would be to programmatically ban a user from all our rooms
